File: Java/백준/Silver/2485. 가로수/가로수.java
import java.io.*;
import java.util.*;

public class Main {
    static int gcd(int a, int b) {
        while (b != 0) {
            int t = a % b;
            a = b;
            b = t;
        }
        return a;
    }

    public static void main(String[] args) throws Exception {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        int n = Integer.parseInt(br.readLine().trim());

        int[] pos = new int[n];
        for (int i = 0; i < n; i++) pos[i] = Integer.parseInt(br.readLine().trim());
        Arrays.sort(pos);

        int[] diffs = new int[n - 1];
        for (int i = 1; i < n; i++) diffs[i - 1] = pos[i] - pos[i - 1];

        int g = diffs[0];
        for (int i = 1; i < diffs.length; i++) g = gcd(g, diffs[i]);

        long ans = 0;
        for (int d : diffs) ans += (d / g) - 1;

        System.out.println(ans);
    }
}

File: Python/백준/Silver/2485. 가로수/가로수.py
import sys
from math import gcd

input = sys.stdin.readline

def main():
    n = int(input().strip())
    pos = [int(input().strip()) for _ in range(n)]
    pos.sort()

    diffs = [pos[i] - pos[i - 1] for i in range(1, n)]
    g = diffs[0]
    for d in diffs[1:]:
        g = gcd(g, d)

    ans = 0
    for d in diffs:
        ans += (d // g) - 1

    print(ans)

if __name__ == "__main__":
    main()
